File: homework-management-java/src/com/company/validation/StudentValidator.java
package com.company.validation;
import com.company.domain.Student;

public class StudentValidator implements Validator<Student> {
    public void validate(Student student) throws ValidationException {
        if (student.getID() == null || student.getID().equals("") || Integer.parseInt(student.getID()) < 1) {
            throw new ValidationException("ID invalid! \n");
        }
        if (student.getNume() == null || student.getNume().equals("")) {
            throw new ValidationException("Nume invalid! \n");
        }
        if (student.getGrupa() <= 110 || student.getGrupa() >= 938) {
            throw new ValidationException("Grupa invalida! \n");
        }
    }
}

File: homework-management-csharp/LAB9-2/validation/StudentValidator.cs
using LAB9_2.domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace LAB9_2.validation
{
    class StudentValidator : IValidator<Student>
    {
        public void Validate(Student student)
        {
            if (student.Id == null || student.Id.Equals("")) throw new ValidationException("ID invalid! \n");
            if (student.Nume == null || student.Nume.Equals("")) throw new ValidationException("Nume invalid! \n");
            if (student.Grupa <= 110 || student.Grupa >= 938) throw new ValidationException("Grupa invalida! \n");
        }
    }
}
